### PR TITLE
fix(select): make aria compliant, read value in screen readers

### LINKED ISSUE #3859: md-select role should be listbox
md-select has an ARIA role of "combobox", but it's not actually a combobox; it's a listbox. A combobox has a text field for the user to type in new items.

> A presentation of a select; usually similar to a textbox where users can type ahead to select an option, or type to enter arbitrary text as a new item in the list.

-- http://www.w3.org/TR/wai-aria/roles#combobox

A combobox is required to have a child element that has role "textbox".

I think listbox is more appropriate:

> A widget that allows the user to select one or more items from a list of choices.

-- http://www.w3.org/TR/wai-aria/roles#listbox

### LINKED ISSUE #3891: md-select doesn't read currently selected item in screen readers
A native `<select>` reads aloud the currently selected item when you focus it. `<md-select>` does not. Reproduced in JAWS/Firefox and Chromevox.

This can be reproduced on the [demo page](https://material.angularjs.org/latest/#/demo/material.components.select).

### LINKED ISSUE #4914: md-select fails chrome accessibility dev tools audit
Assuming that https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb?hl=en is installed...

Go to https://material.angularjs.org/0.11.1/#/demo/material.components.select, open dev tools, go to audit tab and run an audit on that page.

Notice that every instance of md-select throws:
[Severe] Elements with ARIA roles must ensure required owned elements are present (6)

Perhaps it's expecting the aria roles to look more like this:
http://www.w3.org/TR/wai-aria/roles#combobox

### LINKED ISSUE #4977: mdSelect options must be rendered at load time, not just on interaction
In working on #3859 and #4914, I came across an issue that has existed for a while (I've mentioned it to @rschmukler a few times): 

The options for an mdSelect are not in the DOM at all until you click or use the keyboard to activate it. For screen readers, this means a relationship cannot be established between the select and the options it controls, causing an error in Chrome's A11y Developer Tools:
```
[Severe] Elements with ARIA roles must ensure required owned elements are present (1)
```
Here is the part of the ARIA spec this error refers to: http://www.w3.org/TR/wai-aria/roles#mustContain

Furthermore, when the options are rendered, they go into a separate part of the DOM. We can bind a relationship with `aria-owns`, however, that attribute also requires the matching element to be rendered at load time. When I went to do the work required for #3859, I got even more errors in the Chrome A11y audit stemming from this issue. So the problem isn't going away. 

Recommended fix: inject the options on page load so the required ARIA relationships can be established. We can then test `aria-owns` on mdSelect and decide if the options are accessible enough when rendered in a separate part of the DOM.

For async, we could put in at least one option to start (like a placeholder). I'm not sure how accessible this will be, since it might not refresh the accessibility tree reliably, but it's worth a try.

### PR DESCRIPTION
**Pull request by rschmukler**
closes #3891, closes #4914, closes #4977, closes #3859

### CODE REVIEW

File: src/components/select/select.js [hunk 1]
```diff
@@ -290,15 +293,17 @@ function SelectDirective($mdSelect, $mdUtil, $mdTheming, $mdAria, $compile, $par
...
-        setAriaLabel();
+        initAriaLabel();
         syncLabelText();
+        syncAriaLabel();
       });
 
-      function setAriaLabel() {
-        var labelText = element.attr('placeholder');
+      function initAriaLabel() {
+        var labelText = element.attr('aria-label') || element.attr('placeholder');
```
Comment (jelbourn) [inline src/components/select/select.js]: Will this throw spurious errors if using `aria-labelledby` ?

File: src/components/select/select.js [hunk 2]
```diff
@@ -398,18 +409,15 @@ function SelectDirective($mdSelect, $mdUtil, $mdTheming, $mdAria, $compile, $par
...
 
       // Create a fake select to find out the label value
-      function createSelect() {
-        selectContainer = angular.element(selectTemplate);
-        var selectEl = selectContainer.find('md-select-menu');
-        selectEl.data('$ngModelController', ngModelCtrl);
-        selectEl.data('$mdSelectController', mdSelectCtrl);
-        selectScope = scope.$new();
-        $mdTheming.inherit(selectContainer, element);
+      function getSelect() {
```
Comment (jelbourn) [inline src/components/select/select.js]: How about `getNativeSelect` ?

File: src/components/select/select.js [hunk 3]
```diff
@@ -879,14 +891,13 @@ function SelectProvider($$interimElementProvider) {
       }
 
       /**
-       * Detach the element and cleanup prior changes
+       * Clean the element up to a closed state
```
Comment (jelbourn) [inline src/components/select/select.js]: ...up to a closed state?

File: src/components/select/select.js (hunk 2 above)
Comment (jelbourn) [inline src/components/select/select.js]: It's weird to have a function that starts with `get` that doesn't return anything.

Comment (rschmukler) [inline src/components/select/select.js]: It's not a native select. It's just finding the element inside itself that represents the select menu. Maybe `getSelectMenu`

File: src/components/select/select.js (hunk 3 above)
Comment (rschmukler) [inline src/components/select/select.js]: Will fix.

File: src/components/select/select.js (hunk 2 above)
Comment (jelbourn) [inline src/components/select/select.js]: Ah, I was thrown off by the comment. Is it still accurate?

**Comment by jelbourn** [comment]
Overall LGTM aside from minor naming nitpicks.

**Pull request closed without merging**